packages/node-opcua-end2end-test/test/end_to_end/test_e2e_1170.ts

```diff
@@ -0,0 +1,49 @@
+const async = require("async");
+const should = require("should");
+const opcua = require("node-opcua");
+const OPCUAClient = opcua.OPCUAClient;
+const UserTokenPolicy = opcua.UserTokenPolicy;
+const { perform_operation_on_client_session } = require("../../test_helpers/perform_operation_on_client_session");
+// require("node-opcua-service-session").UserNameIdentityToken;
+
+const describe = require("node-opcua-leak-detector").describeWithLeakDetector;
+describe("Testing bug #1170", function () {
+    const port = 1170;
+    let server, client;
+    let endpointUrl;
+    let session;
+    before(async () => {
+        server = new opcua.OPCUAServer({
+            port,
+            serverInfo: {
+                productUri: "Mini NodeOPCUA-Server"
+            }
+        });
+
+        await server.start();
+
+        let endpoints = server._get_endpoints();
+        endpointUrl = endpoints[0].endpointUrl;
+    });
+    after(function (done) {
+        server.shutdown(done);
+    });
+
+    beforeEach(async () => {
```

Comment:
As we have only one test, there is no need for this beforeEach/afterEach block, we could also eliminate the variables at the describe scope

````javascript
it("server createSession should expose endpoints with correct productURI ", async()=> {
   const client = OPCUAClient.create({});
    const serverEndpoints = await client.withSessionAsync(async (session)=>{
            return session.serverEndpoints;
    });
    serverEndpoints[0].server.productUri).should.eql("Mini NodeOPCUA-Server");
}};

````

Reply:
You're right, I've changed